servicex_app/servicex_app/resources/users/token_refresh.py

```diff
@@ -27,19 +27,19 @@
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 from flask_jwt_extended import (create_access_token, decode_token, get_jwt,
-                                get_jwt_identity, jwt_required)
+                                jwt_required)
 from flask_restful import Resource
 from servicex_app.models import UserModel
 
 
 class TokenRefresh(Resource):
     @jwt_required(refresh=True)
     def post(self):
-        user = UserModel.find_by_sub(get_jwt_identity())
         claims = get_jwt()
+        user = UserModel.find_by_email(claims['sub'])
```

Comment:
So once the JWT is unpacked, the `sub` is the email address now?

Reply:
Yes. In principle then, if someone changes OIDC providers but the emails don't change, then the tokens remain valid. We could honestly remove the `sub` field from the database but I was reluctant to do that just yet.

Reply:
Just double checking - the 'sub' is the user's email in this jwt?